src/main/java/dk/kb/storage/storage/DsStorage.java

```diff
@@ -365,6 +381,46 @@ public ArrayList<DsRecordDto > getModifiedAfterParentsOnly(String origin, long m
         return records; 
     }
 
+    
+
+    /**
+     * <p>
+     * Get a list of records after a given mTime. The records will only have fields
+     * id,mTime,referenceid and kalturaid defined 
+     * </p>
+     *
+     *@param origin The origin to fetch records drom
+     *@param mTime only fetch records with mTime larger that this
+     *@param batchSize Number of maximum records to return
+     *
+     * @return List of records only have fields id,mTime,referenceid and kalturaid
+     */
+    public ArrayList<DsRecordReferenceIdDto> getReferenceIds(String origin, long mTime, int batchSize) throws Exception {
+
+        if (batchSize <1 || batchSize > 100000) { //No doom switch
+            throw new Exception("Batchsize must be in range 1 to 100000");          
+        }
```

Comment:
This could be an InvalidArgumentServiceException

Reply:
correct